xmake/modules/detect/tools/nvcc/has_flags.lua

```diff
@@ -31,7 +31,7 @@ function _islinker(flags, opt)
   
     -- the flags is "-Wl,<arg>" or "-Xlinker <arg>"?
     local flags_str = table.concat(flags, " ")
-    if flags_str:startswith("-Wl,") or flags_str:startswith("-Xlinker ") then
+    if flags_str:startswith("-Wl,") or flags_str:startswith("-Xlinker ") flags_str:startswith("-Xcompiler ") then
         return true
```

Comment:
Here is the invalid grammar, we need add `or` keyword, e.g. `or flags_str:startswith("-Xcompiler ")`.

`-Xcompiler xx` is not linker argument options, it is only for compiler. We should not judge it in `function _islinker(flags, opt)`.